### PR TITLE
cluster_upgrade: skip host upgrades without anything to update

### PR DESCRIPTION
**Pull request by mnecas**
Issue: When doing cluster_upgrade on the host without any updates the host upgrade is not started. This creates a problem because we are waiting for the `HOST_UPGRADE_FAILED` `HOST_UPGRADE_FINISHED` `HOST_UPGRADE_FINISHED_AND_WILL_BE_REBOOTED` events, if the upgrade is not started the events are not created so we are waiting for something that will never be there. By default from UI it should timeout after 60 min.

### CODE REVIEW

File: plugins/modules/ovirt_host.py [hunk 1]
```diff
@@ -610,38 +610,44 @@ def main():
...
-                            # 166: No available host was found to migrate VM
-                            search='type=65 or type=140 or type=166 or type=17',
-                        ) if host.name in event.description
-                    ]) > 0
-                ),
-                reboot=module.params['reboot_after_upgrade'],
+            # List all update check events and check if we some updates were found
+            no_upgrade = events_service.list(
+                from_=int(last_event.id),
+                search='type=885 and message="*no updates found*" and host.name=%s' % host.name,
```
Comment (mwperina) [inline plugins/modules/ovirt_host.py:616]: When there are no updates available, then HostUpgradeCommand failes with specific message:

https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/hostdeploy/UpgradeHostCommand.java#L50
https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/validator/UpgradeHostValidator.java#L45

Wouldn't it be easier to check this result?

Comment (mnecas) [reply plugins/modules/ovirt_host.py]: Was able to find middle ground... now it will skip the wait if the update is not available

**Review by mwperina — APPROVED** [review]
+1

File: plugins/modules/ovirt_host.py (hunk 1 above)
Comment (mwperina) [reply plugins/modules/ovirt_host.py]: Great!

**Pull request merged**